src/NServiceBus.Core/Sagas/SagaMetadata.cs

```diff
@@ -227,9 +227,29 @@ public static SagaMetadata Create(Type sagaType, IEnumerable<Type> availableType
                 }
             }
 
+            var invalidEntityProperties = sagaEntityType.GetProperties().Where(IsWritableNonPublic).ToList();
+            if (invalidEntityProperties.Count > 0)
+            {
+                var propertyNames = string.Join(", ", invalidEntityProperties.Select(p => $"'{p.Name}'"));
+                var writableDataPropertiesMustBePublic = $"Writable properties of '{sagaEntityType}' must be public: {propertyNames}";
+                throw new Exception(writableDataPropertiesMustBePublic);
```

Comment:
I'm not sure that we can throw an exception here in a minor. Is it possible that someone has a non-publicly writeable saga data property on an existing implementation and that this is not a mistake?

Reply:
that is 100% a mistake and it'd result in problems. As per connected issue, the persistence mechanism might write the data into storage but won't be able to read it back. It was discussed in the original issue [here](https://github.com/Particular/NServiceBus/issues/5103).